### PR TITLE
feature/#450 added get entities by their config id

### PR DESCRIPTION
**Pull request by toan-quach**
#450

### CODE REVIEW

File: src/taipy/core/_manager/_manager.py [hunk 1]
```diff
@@ -122,3 +122,10 @@ def _export(cls, id: str, folder_path: Union[str, pathlib.Path]):
         Export an entity.
         """
         return cls._repository._export(id, folder_path)
+
+    @classmethod
+    def _get_all_entities_by_config_id(cls, config_id: str) -> List[EntityType]:
```
Comment (jrobinAV) [inline src/taipy/core/_manager/_manager.py]: Maybe we can simplify the name. Here are a few proposal.
`_get_by_config_id` (my preference)
`_get_all_by_config_id`
`_get_entities_by_config_id`

File: src/taipy/core/_repository/_repository.py [hunk 2]
```diff
@@ -135,6 +135,20 @@ def _export(self, entity_id: str, folder_path: Union[str, pathlib.Path]):
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def _get_all_entities_by_config_id(cls, config_id: str) -> List[Entity]:
+        """
+        Get the entities by its cofig id.
```
Comment (jrobinAV) [inline src/taipy/core/_repository/_repository.py]:
```suggestion
        Get the entities by its config id.
```

File: src/taipy/core/_repository/_repository.py [hunk 3]
```diff
@@ -135,6 +135,20 @@ def _export(self, entity_id: str, folder_path: Union[str, pathlib.Path]):
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def _get_all_entities_by_config_id(cls, config_id: str) -> List[Entity]:
```
Comment (jrobinAV) [inline src/taipy/core/_repository/_repository.py]: Same remark on the name.

File: src/taipy/core/taipy.py [hunk 4]
```diff
@@ -639,3 +639,29 @@ def get_cycles_scenarios() -> Dict[Optional[Cycle], List[Scenario]]:
...
+    if entities := _ScenarioManagerFactory._build_manager()._get_all_entities_by_config_id(config_id):
+        return entities
+    if entities := _PipelineManagerFactory._build_manager()._get_all_entities_by_config_id(config_id):
+        return entities
+    if entities := _TaskManagerFactory._build_manager()._get_all_entities_by_config_id(config_id):
+        return entities
+    if entities := _DataManagerFactory._build_manager()._get_all_entities_by_config_id(config_id):
+        return entities
+    if entities := _JobManagerFactory._build_manager()._get_all_entities_by_config_id(config_id):
+        return entities
```
Comment (jrobinAV) [inline src/taipy/core/taipy.py]: Jobs don't have config_id.

```suggestion
```

**Review by jrobinAV — CHANGES_REQUESTED** [review]

File: src/taipy/core/_repository/_filesystem_repository.py [hunk 5]
```diff
@@ -136,6 +136,10 @@ def _delete_by(self, attribute: str, value: str, version_number: Optional[str] =
     def _search(self, attribute: str, value: str, version_number: Optional[str] = None) -> Optional[Entity]:
         return next(self.__search(attribute, value, version_number), None)
 
+    def _get_all_entities_by_config_id(self, config_id: str) -> List[Entity]:
+        files = filter(lambda f: config_id in f.name, self.dir_path.iterdir())
+        return [self.__to_entity(f, retry=Config.global_config.read_entity_retry or 0) for f in files]  # type: ignore
```
Comment (trgiangdo) [inline src/taipy/core/_repository/_filesystem_repository.py]: I guess the `# type: ignore` is related to https://github.com/Avaiga/taipy-core/issues/550 (Config has no global_config attribute) right?

Let's keep this in mind so we can remove this in the future.

**Comment by jrobinAV** [comment]
Only the remark on file `src/taipy/core/taipy.py` is blocking. The others are minor non blocking remarks.

File: src/taipy/core/_repository/_sql_repository.py [hunk 6]
```diff
@@ -167,15 +176,15 @@ def _get_by_configs_and_owner_ids(self, configs_and_owner_ids):
...
+            entry = self.__get_entity_by_config_and_owner(config.id, owner)
             if entry:
                 entity = self.__to_entity(entry)
                 key = config, owner
                 res[key] = entity
 
         return res
 
-    def __get_entities_by_config_and_owner(
+    def __get_entity_by_config_and_owner(
```
Comment (trgiangdo) [inline src/taipy/core/_repository/_sql_repository.py]:
```suggestion
    def __get_model_by_config_and_owner(
```
This method returns a model right?

**Review by trgiangdo — COMMENTED** [review]
I have some really minor comments but nothing directly related to the issue though.

File: src/taipy/core/_manager/_manager.py (hunk 1 above)
Comment (toan-quach) [reply src/taipy/core/_manager/_manager.py]: I'm going with `_get_by_config_id` :D

File: src/taipy/core/_repository/_filesystem_repository.py (hunk 5 above)
Comment (toan-quach) [reply src/taipy/core/_repository/_filesystem_repository.py]: I believe it's caused by `__to_entity` as it returns `Optional[Entity]`. I have replaced this snippet with a for loop and an if condition and it solved the type: ignore issue :D

File: src/taipy/core/_repository/_sql_repository.py [hunk 7]
```diff
@@ -166,7 +166,7 @@ def _get_all_entities_by_config_id(self, config_id: str) -> List[Entity]:
         return [self.__to_entity(entity_model) for entity_model in entity_models]
 
     def _get_by_config_and_owner_id(self, config_id: str, owner_id: Optional[str]) -> Optional[Entity]:
-        entity = self.__get_entity_by_config_and_owner(config_id, owner_id)
+        entity = self.__get_model_by_config_and_owner(config_id, owner_id)
```
Comment (jrobinAV) [inline src/taipy/core/_repository/_sql_repository.py]:
```suggestion
        model = self.__get_model_by_config_and_owner(config_id, owner_id)
```

File: src/taipy/core/_repository/_sql_repository.py [hunk 8]
```diff
@@ -166,7 +166,7 @@ def _get_all_entities_by_config_id(self, config_id: str) -> List[Entity]:
...
         return self.__to_entity(entity)
```
Comment (jrobinAV) [inline src/taipy/core/_repository/_sql_repository.py]:
```suggestion
        return self.__to_entity(model)
```

**Review by jrobinAV — DISMISSED** [review]

**Review by jrobinAV — APPROVED** [review]

**Pull request merged**